HACKING.rst

```diff
@@ -86,18 +86,129 @@ You can also see all the environments by simply running ``tox list``
 Running ``tox run -m format`` and ``tox run -m lint`` before committing code is
 recommended.
 
-Code Style
-==========
-
 Commits
-#######
+-------
+
+Commit messages are based on the `conventional commit`_ style::
+
+  <type>(<optional scope>): <description>
+
+  <optional body>
+
+  <optional footer>
+
+The commit is divided into three sections: a header, body, and footer.
+
+Header
+======
+
+The header is required and consists of three subsections: a type,
+optional scope, and description. The header must be 72 characters or less.
+
+Types
+#####
+
+``build``
+"""""""""
+
+Commits that affect the build of an application or library.
+
+This includes dependency updates, which should use the ``deps`` scope
+(``build(deps):``).
+
+``chore``
+"""""""""
+
+Miscellaneous commits that don't fit into any other type.
+
+Examples include:
+
+* development dependency updates
+* edits to a comment or docstring
+* type changes
+* accommodating a developer-facing deprecation warning
+* many *small* fixes for an existing PR
+
+``ci``
+""""""
+
+Commits that affect the CI/CD pipeline.
+
+``docs``
+""""""""
```

Comment:
A hill I'm absolutely not willing to die on, but wanted to bring up anyway:

For things that don't change the meaning of the documentation (e.g. fixing a typo, making `sphinx-lint` happy, conformance with the [style guide](https://docs.ubuntu.com/styleguide/en/)), it might be better to use `style(docs)`.

Reasoning: When doing a release, `docs` changes are things I'll look at since new documentation is also a feature, but `style` is stuff I'll generally scan over since they shouldn't affect the behaviour. (And fixing a typo isn't a bugfix since people wull steal undirstend the meening.

Reply:
I like it.  I can add a note about this scenario.